From c44ae3bb30b0a12672123f4184f599fb0b5904a7 Mon Sep 17 00:00:00 2001
From: Rex Dieter <rdieter@gmail.com>
Date: Fri, 29 Jun 2018 08:43:14 -0500
Subject: [PATCH] support python-3.7

---
 cmake/FindPythonLibs3.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindPythonLibs3.cmake b/cmake/FindPythonLibs3.cmake
index e792dcfa..48e6c8eb 100644
--- a/cmake/FindPythonLibs3.cmake
+++ b/cmake/FindPythonLibs3.cmake
@@ -5,7 +5,7 @@ set(PYTHONLIBS3_DEFINITIONS ${PYTHONLIBS3_CFLAGS_OTHER})
 find_path(PYTHONLIBS3_INCLUDE_DIR Python.h
           HINTS ${PYTHONLIBS3_INCLUDEDIR} ${PYTHONLIBS3_INCLUDE_DIRS})
 
-find_library(PYTHONLIBS3_LIBRARY NAMES python3.2m python3.3m python3.4m python3.5m python3.6m
+find_library(PYTHONLIBS3_LIBRARY NAMES python3.2m python3.3m python3.4m python3.5m python3.6m python3.7m
              HINTS ${PYTHONLIBS3_LIBDIR} ${PYTHONLIBS3_LIBRARY_DIRS} )
 
 set(PYTHONLIBS3_LIBRARIES ${PYTHONLIBS3_LIBRARY} )
-- 
2.17.1

