From cc26da79e51f4826a9b34ae034fc40704b2748e9 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Mon, 21 Jan 2013 23:36:06 +0100
Subject: [PATCH 3/4] Actually make my check work

---
 cmake/FindR.cmake | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/cmake/FindR.cmake b/cmake/FindR.cmake
index f0dcec4..11f4aff 100644
--- a/cmake/FindR.cmake
+++ b/cmake/FindR.cmake
@@ -10,6 +10,7 @@
 # find the R binary
 FIND_PROGRAM(R_EXECUTABLE R)
 
+SET(ABORT_CONFIG FALSE)
 IF(R_EXECUTABLE)
 
   # find R_HOME
@@ -77,7 +78,8 @@ IF(R_EXECUTABLE)
         # and compiling on windows is hard enough even without requiring libgfortran, too.
         SET(R_LIBRARIES ${R_LIBRARIES} gfortran)
       ELSE (GFORTRAN_LIBRARY)
-        SET(R_FOUND false)
+        MESSAGE(STATUS "gfortran is needed for Rlapack but it could not be found")
+        SET(ABORT_CONFIG TRUE)
       ENDIF (GFORTRAN_LIBRARY)
     ENDIF(NOT WIN32)
   ENDIF(NOT R_LAPACK_LIBRARY)
@@ -99,8 +101,12 @@ IF(R_EXECUTABLE)
 
 ENDIF( R_EXECUTABLE )
 
-INCLUDE(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(R  DEFAULT_MSG
+IF(ABORT_CONFIG)
+  SET(R_FOUND FALSE)
+ELSE(ABORT_CONFIG)
+  INCLUDE(FindPackageHandleStandardArgs)
+  FIND_PACKAGE_HANDLE_STANDARD_ARGS(R  DEFAULT_MSG
                                   R_EXECUTABLE R_INCLUDE_DIR R_R_LIBRARY)
 
-MARK_AS_ADVANCED(R_INCLUDE_DIR R_R_LIBRARY R_LAPACK_LIBRARY R_BLAS_LIBRARY)
+  MARK_AS_ADVANCED(R_INCLUDE_DIR R_R_LIBRARY R_LAPACK_LIBRARY R_BLAS_LIBRARY)
+ENDIF(ABORT_CONFIG)
-- 
1.8.1

