--- cantor-21.04.3/src/backends/julia/juliaserver/juliaserver.cpp.old	2021-07-06 04:00:21.000000000 +0900
+++ cantor-21.04.3/src/backends/julia/juliaserver/juliaserver.cpp	2021-08-25 21:51:00.236079959 +0900
@@ -106,7 +106,9 @@ void JuliaServer::runJuliaCommand(const
 
     if (jl_exception_occurred()) { // If exception occurred
         // Show it to user in stderr
-#if QT_VERSION_CHECK(JULIA_VERSION_MAJOR, JULIA_VERSION_MINOR, 0) >= QT_VERSION_CHECK(1, 1, 0)
+#if QT_VERSION_CHECK(JULIA_VERSION_MAJOR, JULIA_VERSION_MINOR, 0) >= QT_VERSION_CHECK(1, 7, 0)
+        jl_value_t *ex = jl_current_task->ptls->previous_exception;
+#elif QT_VERSION_CHECK(JULIA_VERSION_MAJOR, JULIA_VERSION_MINOR, 0) >= QT_VERSION_CHECK(1, 1, 0)
         jl_value_t *ex = jl_get_ptls_states()->previous_exception;
 #else
         jl_value_t *ex = jl_exception_in_transit;
